src/pages/Home/Home.page.jsx

```diff
@@ -1,39 +1,34 @@
-import React, { useRef } from 'react';
-import { Link, useHistory } from 'react-router-dom';
+import React from 'react';
+import Header from '../../components/Header';
+import VideoCard from '../../components/VideoCard';
 
-import { useAuth } from '../../providers/Auth';
-import './Home.styles.css';
+import { GridContainer, Title } from './styled';
+import { items } from '../../mock/youtube-videos-mock.json';
 
-function HomePage() {
-  const history = useHistory();
-  const sectionRef = useRef(null);
-  const { authenticated, logout } = useAuth();
-
-  function deAuthenticate(event) {
-    event.preventDefault();
-    logout();
-    history.push('/');
-  }
+const getVideoCardsFromList = (list) => {
+  return list.map((i) => {
+    const { snippet } = i;
+    const { title, description, thumbnails } = snippet;
+    return (
+      <VideoCard
+        title={title}
+        description={description}
+        thumbnail={thumbnails.high.url}
+      />
+    );
+  });
+};
 
+const HomePage = () => {
   return (
-    <section className="homepage" ref={sectionRef}>
-      <h1>Hello stranger!</h1>
-      {authenticated ? (
-        <>
-          <h2>Good to have you back</h2>
-          <span>
-            <Link to="/" onClick={deAuthenticate}>
-              ← logout
-            </Link>
-            <span className="separator" />
-            <Link to="/secret">show me something cool →</Link>
-          </span>
-        </>
-      ) : (
-        <Link to="/login">let me in →</Link>
-      )}
-    </section>
+    <>
+      <Header />
+      <Title>Welcome to my minichallenge :)</Title>
+      <GridContainer>{getVideoCardsFromList(items)}</GridContainer>
```

Comment:
Here you can do something like:
```jsx
{items.map((item) => {
  const { title, description, thumbnails } = item.snippet;
  return (
    <VideoCard title={title} description={description} thumbnail={thumbnail} key={someUniquevalue} />
  );
})}
```

You can use the function, but it is not that common, also, when looping throughout an array of items, remember that react requires the `key` prop to work properly, if you omit the `key` you may get some bugs that are not easy to identify

Reply:
Well, I made it this way just to have more test coverage. 😅  I didn't like how it looked either. I'll correct it. :)